sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -2018,21 +2019,25 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
 
   Seq("a b", "a:b", "a%b", "a,b").foreach { specialChars =>
     test(s"data source table:partition column name containing $specialChars") {
+      // On Windows, it looks colon in the file name is illegal by default. See
+      // https://support.microsoft.com/en-us/help/289627
+      assume(!Utils.isWindows || specialChars != "a:b")
```

Comment:
It seems `:` is not allowed on Windows by default - https://support.microsoft.com/en-us/help/289627

**Windows**

```scala
scala> new java.io.File("a").mkdirs()
res0: Boolean = true

scala> new java.io.File("a:").mkdirs()
res1: Boolean = false

scala> new java.io.File("a:b").mkdirs()
res2: Boolean = false

scala> new java.io.File("a\\:b").mkdirs()
res3: Boolean = false
```

**Mac**

```scala
scala> new java.io.File("a:b").mkdirs()
res0: Boolean = true
```